src/plugins/intel_gpu/src/graph/impls/ocl/kv_cache.cpp

```diff
@@ -65,40 +67,74 @@ struct kv_cache_impl : multi_stage_primitive<kv_cache> {
     using dq_kernel_selector_t = kernel_selector::dynamic_quantize_kernel_selector;
     using dq_kernel_params_t = kernel_selector::dynamic_quantize_params;
 
+    using scatter_kernel_selector_t = kernel_selector::scatter_elements_update_kernel_selector;
+    using scatter_kernel_params_t = kernel_selector::scatter_elements_update_params;
+
     DECLARE_OBJECT_TYPE_SERIALIZATION(cldnn::ocl::kv_cache_impl)
 
+    // Constructor that initializes stage indices based on kernels
+    kv_cache_impl(const std::vector<kernel_selector::kernel_data>& kernels_data) : parent(kernels_data) {
+        initialize_stage_indices();
+    }
+
     std::unique_ptr<primitive_impl> clone() const override {
         return make_deep_copy<kv_cache_impl, kernel_params_t>(*this);
     }
 
-    const size_t concat_stage = 0;
-    const size_t beam_table_stage = 1;
-    const size_t dq_stage = 2;
-    const size_t scale_concat_stage = 3;
-    const size_t zp_concat_stage = 4;
+    size_t reorder_stage = -1;
```

Comment:
This name `reorder` is misleading because it is widely used as "re-layout the physical format while keeping logic format intact" in GPU. What about `update_stage` or `scatter_update_stage`?